csharp/src/Microsoft.ML.OnnxRuntime/OnnxRuntime.shared.cs

```diff
@@ -85,11 +87,48 @@ internal static IntPtr Handle
         #region public methods
 
         /// <summary>
-        /// Returns an instance of OrtEnv
+        /// Returns an instance of OrtEnv, with customized log level
         /// It returns the same instance on every call - `OrtEnv` is singleton
         /// </summary>
+        /// <param name="logLevel">LogLevel instance to be used for OrtEnv initialization</param>
         /// <returns>Returns a singleton instance of OrtEnv that represents native OrtEnv object</returns>
-        public static OrtEnv Instance() { return _instance.Value; }
+        public static OrtEnv Instance(LogLevel logLevel)
+        {
+            if (_instance == null)
```

Comment:
Just wondering is this code thread-safe?

The previous code `private static readonly Lazy<OrtEnv> _instance = new Lazy<OrtEnv>(()=> new OrtEnv())` implicitly uses LazyThreadSafetyMode.ExecutionAndPublication. How about here?

Reply:
This is not thread-safe